gdk/src/display.rs

```diff
@@ -0,0 +1,70 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use crate::Display;
+use glib::object::IsA;
+use glib::ObjectExt;
+
+#[derive(Debug, PartialEq, Eq, Ord, PartialOrd)]
+pub enum Backend {
+    Wayland,
+    X11,
+    Win32,
+    MacOS,
+    Broadway,
+}
+
+impl Backend {
+    // rustdoc-stripper-ignore-next
+    /// Equivalent to the C macro `GDK_IS_WAYLAND_DISPLAY`
+    #[doc(alias = "GDK_IS_WAYLAND_DISPLAY")]
+    pub fn is_wayland(&self) -> bool {
+        matches!(self, Self::Wayland)
+    }
+
+    // rustdoc-stripper-ignore-next
+    /// Equivalent to the C macro `GDK_IS_X11_DISPLAY`
+    #[doc(alias = "GDK_IS_X11_DISPLAY")]
+    pub fn is_x11(&self) -> bool {
+        matches!(self, Self::X11)
+    }
+
+    // rustdoc-stripper-ignore-next
+    /// Equivalent to the C macro `GDK_IS_WIN32_DISPLAY`
+    #[doc(alias = "GDK_IS_WIN32_DISPLAY")]
+    pub fn is_win32(&self) -> bool {
+        matches!(self, Self::Win32)
+    }
+
+    // rustdoc-stripper-ignore-next
+    /// Equivalent to the C macro `GDK_IS_MACOS_DISPLAY`
+    #[doc(alias = "GDK_IS_MACOS_DISPLAY")]
+    pub fn is_macos(&self) -> bool {
+        matches!(self, Self::MacOS)
+    }
+
+    // rustdoc-stripper-ignore-next
+    /// Equivalent to the C macro `GDK_IS_BROADWAY_DISPLAY`
+    #[doc(alias = "GDK_IS_BROADWAY_DISPLAY")]
+    pub fn is_broadway(&self) -> bool {
+        matches!(self, Self::Broadway)
+    }
+}
+
+pub trait DisplayExtManual: 'static {
+    // rustdoc-stripper-ignore-next
+    /// Get the currently used display backend
+    fn backend(&self) -> Backend;
+}
+
+impl<O: IsA<Display>> DisplayExtManual for O {
+    fn backend(&self) -> Backend {
+        match self.as_ref().type_().name() {
+            "GdkWaylandDisplay" => Backend::Wayland,
+            "GdkX11Display" => Backend::X11,
+            "GdkMacosDisplay" => Backend::MacOS,
```

Comment:
```suggestion
            "GdkQuartzDisplay" => Backend::MacOS,
```

Please also double-check the others

Reply:
Checking from https://gitlab.gnome.org/GNOME/gtk/-/tree/gtk-3-24/gdk, it seems to be the only one (took me a while to understand I was not looking at the right version...).